### ISSUE TITLE
Replace the magic getters/setters with non-magic ones

### ISSUE THREAD

**Issue by oliverklee** [mjaschen/collmex#130]
The `AbstractType::__get` and `AbstractType::__set` do not allow static analysis for cases where they are used. I'd like to deprecate them and replace them with explicit `get` and `set` methods together with explicitly-typed `getAsString`, `setAsBoolean` etc. methods.

@mjaschen What do you think?

**Comment by oliverklee** [comment]
(And also, Prophecy can only stub/mock existing methods, not magic method calls.)

**Comment by mjaschen** [comment]
Type properties would be accessed like `$type->getAsString('client_id')` or `$type->get('client_id`)`, right?

**Comment by oliverklee** [comment]
Yes, exactly: with `getAsString` already typed as string, and with `get` with mixed type/untyped.

**Comment by mjaschen** [comment]
Sounds like a good idea.

Maybe we can add specific getters/setters for each field on a per type basis in a later step (`getClientId()`, `setFirstname()`). That'd be a lot of boilerplate code but would also simplify working with the library.

**Comment by oliverklee** [comment]
> Maybe we can add specific getters/setters for each field on a per type basis in a later step 

Sounds good.

Then I propose we move this to post-1.0 (i.e., 2.0 as this will be breaking as well).

**Comment by mjaschen** [comment]
(for later reference)

The Collmex API uses six different data types[1]:

- alpha-numerical (i.e. “strings”)
- numerical, integer: an ordinary integer value
  - also used for IDs: in this case it can be followed by a space and a string which is the readable name for the referenced record. Example: `5` and `5 PayPal` are two representations for the same value. The appended space + string can be omitted when using in a setter. We should make sure to process it when parsing a CSV document though (maybe even expose it in a special getter method).
- numerical, decimal: a decimal value with up to 3 decimals. Collmex returns (and expects) values in the german localization (comma as decimal separator); we should take care of conversions between PHP's float values and Collmex' numerical values in the setters/getters
- money: a decimal value with 2 decimals, same notes as on “numerical, decimal” apply here; maybe we can use [moneyphp/money](https://packagist.org/packages/moneyphp/money) to store money values internally
- date: a date string in the format `YYYYMMDD` or `DD.MM.YYYY`; we should use only the former format when creating API data; dates should stored as `DateTime` internally
- time: a time string in the format `HHMMSS`

[1] <https://www.collmex.de/cgi-bin/cgi.exe?1005,1,help,daten_importieren_datentypen_felder>
